### PR TITLE
Feature logger

### PR DESCRIPTION
**Pull request by akdilsiz**
Fully customizable logger structure added.

### CODE REVIEW

File: gocron_test.go [hunk 1]
```diff
@@ -684,3 +707,25 @@ func TestGetWeekday(t *testing.T) {
...
+		logger.Info("Info")
+	})
+}
+
+func TestScheduleSetLogger(t *testing.T) {
+	sched := NewScheduler()
+	sched.SetLogger(NewCustomLogger())
+	job := sched.Every(1).Second()
+	testJobWithInterval(t, sched, job, 1)
+}
```
Comment (JohnRoesler) [inline gocron_test.go]: could you add a newline?

Comment (akdilsiz) [reply gocron_test.go]: Of course, I added :)

**Comment by Streppel** [comment]
Cool @akdilsiz, thanks! I see that John already reviewed the code, I'll take a look as soon as I get some time and we will merge this probably this weekend

**Comment by akdilsiz** [comment]
@Streppel Actually, it's better. We can work on a separate branch to further develop.

**Comment by Streppel** [comment]
@akdilsiz merged on feature/job-logger. sorry for the long time it took, been pretty busy this beginning of the year. I'll make some commits to this branch as soon as I get some time to invest here!

**Comment by JohnRoesler** [comment]
How is this one coming? Made progress towards getting it into master?

**Comment by Streppel** [comment]
This is still not ready @JohnRoesler, there are some things that could be improved here and there about both implementation and design and some testing as well.

**Pull request merged**